cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/BrokerFailureDetectorTest.java

```diff
@@ -197,4 +228,14 @@ private void restartDeadBroker(int index) {
   private MockTime getMockTime() {
     return new MockTime(0, 100L, TimeUnit.NANOSECONDS.convert(100L, TimeUnit.MILLISECONDS));
   }
+
+  private void detect(AbstractBrokerFailureDetector detector, Queue<Anomaly> anomalies, long start, int count) {
+    if (_useKafkaApi) {
+      detector.run();
+    } else {
```

Comment:
In a real execution, isn't `run()` called regardless of whether `_useKafkaApi` is `true` or not? To better emulate the actual run, shouldn't we always call `detector.run();`?